### PR TITLE
[Integration][idrac_secure_boot]- Added automation for attributes

### PR DESCRIPTION
**Pull request by Saksham-Nautiyal**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

### CODE REVIEW

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

File: tests/integration/targets/idrac_secure_boot/tests/attributes.yaml [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+
+    - name: Verify - Set boot mode to Uefi (normal mode)
+      ansible.builtin.assert:
+        that:
+          - boot_md_uefi.changed
+          - boot_md_uefi.msg == 'Successfully updated the BIOS attributes.'
+
+    - name: Get details by uri.
+      ansible.builtin.uri:
+        url: "https://{{ idrac_ip | ansible.utils.ipwrap
```
Comment (ShivamSh3) [inline tests/integration/targets/idrac_secure_boot/tests/attributes.yaml]: No need of putting ipwrap everywhere, if it is already mentioned in module_defaults.

File: tests/integration/targets/idrac_secure_boot/tests/attributes.yaml [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
+        status_code: 200
+      register: boot_md_uri
+
+    - name: Verify -  Get details by uri.
+      ansible.builtin.assert:
+        that:
+          - boot_md_uefi.changed
+          - 'boot_md_uri.json.Attributes.BootMode == "Uefi"'
+
+    - name: Set boot mode to bios (check mode - changes expected)
```
Comment (singh-lovepreet1) [inline tests/integration/targets/idrac_secure_boot/tests/attributes.yaml]: Change the sequence of tasks. First set to BIOS and then UEFI

File: tests/integration/targets/idrac_secure_boot/tests/invalid_secure_boot.yaml [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+      dellemc.openmanage.idrac_secure_boot:
+        idrac_ip: "{{ idrac_ip }}"
+        idrac_user: "{{ idrac_user }}"
+        idrac_password: "{{ idrac_password }}"
+        validate_certs: false
+        boot_mode: Uefi
+        restart: true
+        restart_type: GracefulRestart
+      register: cm_changes
+
```
Comment (singh-lovepreet1) [inline tests/integration/targets/idrac_secure_boot/tests/invalid_secure_boot.yaml]: Verify from the URI whether force_int_10 is in 'Enabled' state as a pre-requisite

File: tests/integration/targets/idrac_secure_boot/tests/invalid_secure_boot.yaml [hunk 4]
```diff
@@ -0,0 +1,156 @@
+# Dell OpenManage Ansible modules
+# Copyright (C) 2024 Dell Inc. or its subsidiaries. All Rights Reserved.
+
+# GNU General Public License v3.0+ (see COPYING or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+---
+- name: Test that we have an iDRAC host, iDRAC username and iDRAC password
+  ansible.builtin.fail:
```
Comment (singh-lovepreet1) [inline tests/integration/targets/idrac_secure_boot/tests/invalid_secure_boot.yaml]: Rename the file as this covers positive scenarios

**Review by singh-lovepreet1 — DISMISSED** [review]

File: tests/integration/targets/idrac_secure_boot/tests/attributes.yaml (hunk 2 above)
Comment (Saksham-Nautiyal) [reply tests/integration/targets/idrac_secure_boot/tests/attributes.yaml]: done

File: tests/integration/targets/idrac_secure_boot/tests/invalid_secure_boot.yaml (hunk 4 above)
Comment (Saksham-Nautiyal) [reply tests/integration/targets/idrac_secure_boot/tests/invalid_secure_boot.yaml]: done

File: tests/integration/targets/idrac_secure_boot/tests/invalid_secure_boot.yaml (hunk 3 above)
Comment (Saksham-Nautiyal) [reply tests/integration/targets/idrac_secure_boot/tests/invalid_secure_boot.yaml]: done

**Review by rajshekarp87 — APPROVED** [review]

**Review by sachin-apa — APPROVED** [review]

**Review by singh-lovepreet1 — APPROVED** [review]

**Pull request merged**